common4j/src/main/com/microsoft/identity/common/java/providers/oauth2/OAuth2StrategyParameters.java

```diff
@@ -42,4 +43,7 @@ public class OAuth2StrategyParameters {
 
     @Nullable
     private final transient AbstractAuthenticationScheme mAuthenticationScheme;
+
+    @Setter
+    private transient boolean mUsingOpenIdConfiguration;
```

Comment:
This probably better belongs to MicrosoftStsOAuth2Configuration class and a public method isOpenIdConfigUsed() or getOpenIdConfig()
With that MicrosoftStsOAuth2Configuration .getAuthorizationEndpoint() method can be modified to consider this openid configuration before returning result (That method currently does not have any callers, if it had you would have had to make modifications there as well.).

The config object can also then be used to setup the authorization request builder in strategy.

Reply:
I'll add a TODO to consider moving this to MicrosoftStsOAuth2Configuration. I think it would be better to make this change and update the endpoint methods in the same PR, but I'm going to save it for a later PR as this one is decently dense as is and is mainly focused on getting everything wired up.

Reply:
On  a side note, why does the configuration class even have a method for the authorization endpoint, shouldn't that come from the strategy class?